src/v/cloud_storage/materialized_segments.cc

```diff
@@ -140,10 +140,13 @@ ss::future<> materialized_segments::flush_evicted() {
 ss::future<> materialized_segments::run_eviction_loop() {
     // Evict readers asynchronously
```

Comment:
LGTM

At first I wondered if we'd ever end up where `stop()` doesn't call `_eviction_in_flight`, but the end condition of this loop is basically that the condition variable is broken, and we'll always finish emptying the in-flight list before checking the cv again.